components/admin/reports/GenerateReportModal.tsx

```diff
@@ -39,6 +47,63 @@ export default function GenerateReportModal<T>(props: GenerateReportProps<T>) {
   const [endDate, setEndDate] = useState('');
   const [step, setStep] = useState(GenerateReportStep.SelectColumns);
 
+  // Checkbox states
+  const [selectAll, setSelectAll] = useState(false);
+  const [userID, setUserID] = useState(false);
+  const [applicantName, setApplicantName] = useState(false);
+  const [applicantDoB, setApplicantDoB] = useState(false);
+  const [APPNumber, setAPPNumber] = useState(false);
+  const [applicationDate, setApplicationDate] = useState(false);
+  const [paymentMethod, setPaymentMethod] = useState(false);
+  const [feeAmount, setFeeAmount] = useState(false);
+  const [donationAmount, setDonationAmount] = useState(false);
+  const [totalAmount, setTotalAmount] = useState(false);
```

Comment:
A cleaner approach to this would be to use a JavaScript `Set` - we can do this because the API endpoint @j-bunk developed is order-agnostic. We would have something like:
```typescript
const [selectedColumns, setSelectedColumns] = useState<Set<ApplicationsReportColumn.UserId>>(new Set())

// A function that returns an event-handler function that adds/removes the column from selectedColumns
const handleSelectColumn = (column: ApplicationsReportColumn) => (event: SyntheticEvent) => {
  const updatedSelectedColumns = new Set([...selectedColumns])
  event.target.checked ? updatedSelectedColumns.add(column) : updatedSelectedColumns.delete(column)
  setSelectedColumns(updatedSelectedColumns)
}

const handleSelectAllColumns = (event: SyntheticEvent) => {
  const updatedSelectedColumns = event.target.checked ? new Set(<all the columns>) : new Set([])
  setSelectedColumns(updatedSelectedColumns)
}

// Sample checkbox:
<Checkbox
  isChecked={selectedColumns.has(ApplicationsReportColumn.FeeAmount)}
  onChange={handleSelectColumn(ApplicationsReportColumn.FeeAmount)}
>
  Fee Amount
</Checkbox>
```

You can convert a set into an array like so: `[...someSet]`
For disabling the submit button, we just need to check if the set is empty
Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Reply:
Addressed in latest commit!